app/src/main/java/ch/epfl/cs311/wanderwave/model/data/Beacon.kt

```diff
@@ -0,0 +1,38 @@
+package ch.epfl.cs311.wanderwave.model.data
+
+import com.google.firebase.firestore.DocumentSnapshot
+
+data class Beacon(
+
+    /** GUID of the beacon */
+    val id: String,
+
+    /** Location of the beacon */
+    val location: Location,
+
+    /** List of tracks that are broadcast from the beacon */
+    val tracks: List<Track> = listOf<Track>(),
+) {
+  fun toMap(): Map<String, Any> {
+    return hashMapOf("id" to id, "location" to location.toMap(), "tracks" to listOf<Track>())
+  }
+
+  companion object {
+    fun from(document: DocumentSnapshot): Beacon? {
+      return if (document.exists()) {
+        val id = document.id
+        val locationMap = document.get("location") as? Map<String, Any>
+        val latitude = locationMap?.get("latitude") as? Double ?: 0.0
+        val longitude = locationMap?.get("longitude") as? Double ?: 0.0
+        val name = locationMap?.get("name") as? String ?: ""
+        val location = Location(latitude, longitude, name)
+
+        val tracks = listOf<Track>()
```

Comment:
Same here, though I get that it's different here since the DocumentSnapshot does not contain the beacon list... Any ideas on how to do this?

Reply:
@KlaKalma

Reply:
Because it cannot retrieve from the beacon document the track, there are references but not the tracks themself, that need to be fetched, we could add the list of tracks inside the beacon